### DISCUSSION TITLE
Disable automatic bitmap image resizing/cropping

### DISCUSSION THREAD

**Discussion by Frosty-J** [ScratchAddons/ScratchAddons#4630] [Ideas]
Hi all, I've picked up Scratch again for the first time in a very long time, and this extension is great for making it a little more tolerable to use. It's unfortunate the Scratch team doesn't seem especially interested in adding some of the quality-of-life features, not even ones that existed in previous versions.

One issue I'm having with Scratch is the editor's insistence on altering my images. When importing an image, its resolution gets doubled on both axes, e.g. a 32×32 image becomes 64×64 (though is still displayed at 32×32 when the sprite is at 100% size). Additionally, it gets converted from an indexed colour palette to full RGB32. This isn't a huge deal, as normally the images don't add up to a great size anyway, and it's not to difficult to download and extract the project to undo these changes, but it would be nice to avoid. It doesn't seem to mind checksums no longer matching.

A bigger issue is that, when editing the image, the whitespace gets cropped. As Scratch 3.0 has a bug where the edges can get cropped in-game, especially if the origin (1.4 called it the "costume center") isn't centred, preserving at least 1px whitespace on each edge to compensate for that is important to me. It can also be useful to use whitespace for enabling sprites to move off the edge of the screen (the size-changing workaround didn't work in my case due to having a very wide sprite). The editor's cropping can be avoided by manually changing the `rotationCenterX` and `rotationCenterY` values in `project.json`, but it's not the ideal workflow.

I'm aware that Scratch Addons has an "HD image uploads" feature (this probably means https://github.com/ScratchAddons/ScratchAddons/discussions/2085 can be closed?), which packs the image into an SVG to stop Scratch tampering with it, but then the origin crosshair isn't aligned to specific pixels (or half-pixels after Scratch has doubled the image size, depending how you think about it) which is mighty inconvenient for dealing with pixel art. Not convinced it's the most efficient solution either... not that Scratch is for anyone who gives a toss about efficiency.

How feasible would it be to prevent Scratch from changing images without any conversion process? Like how I can upload an MP3 to Scratch and get the exact same MP3 back, except instead of an MP3 it would be a PNG. Obviously Scratch would have to touch the image if you start drawing on it, but it should be able to leave it alone when importing or changing the origin. I'm unfamiliar with what a userscript would have to hook into to try and prevent this.

**Comment by WorldLanguages** [comment]
> Like how I can upload an MP3 to Scratch and get the exact same MP3 back

You actually get a compressed WAV (but sorry this is not the main point of this post)

**Reply by Frosty-J** [reply]
I think only if you edit the sound? Such as reversing it or whatever. I must admit, I was expecting to have to work with mono 22,050Hz IMA ADPCM, as this is how it was in 1.4, but after uploading an MP3 it remains unmodified.

Regardless of whether downloaded as part of an SB3 or as an individual file on the project page, it retains stereo sound and matches my original checksum. I haven't experimented with it a great deal, so it's possible certain metadata or bitrates cause it to get modified.

**Reply by CST1229** [reply]
> I was expecting to have to work with mono 22,050Hz IMA ADPCM, as this is how it was in 1.4

I think in 3.0, the sound conversion when editing keeps the sample rate.

**Comment by Chameleon-Coder** [comment]
That sounds kinda good.....I make small browser games with TurboWarp. So, if I were to make a small easter egg (like, say, a small room that can only be accessed by a very complicated method) that is 20x20, it would become a 40x40. I don't want that.
